tools/generator/dfg/sam/avr_mcu.py

```diff
@@ -0,0 +1,185 @@
+
+    "attiny10":         "t10",
+    "attiny11":         "t11",
+    "attiny12":         "t12",
+    "attiny13":         "t13",
+    "attiny15":         "t15",
+    "attiny1634":       "t1634",
+    "attiny20":         "t20",
+    "attiny2313":       "t2313",
+    "attiny24":         "t24",
+    "attiny25":         "t25",
+    "attiny26":         "t26",
+    "attiny261":        "t261",
+    "attiny28":         "t28",
+    "attiny4":          "t4",
+    "attiny40":         "t40",
+    "attiny4313":       "t4313",
+    "attiny43u":        "t43u",
+    "attiny44":         "t44",
+    "attiny45":         "t45",
+    "attiny461":        "t461",
+    "attiny5":          "t5",
+    "attiny84":         "t84",
+    "attiny85":         "t85",
+    "attiny861":        "t861",
+    "attiny88":         "t88",
+    "attiny9":          "t9",
+
+    "atxmega128a1":     "x128a1",
+    "atxmega128a1revd": "x128a1d",
+    "atxmega128a1u":    "x128a1u",
+    "atxmega128a3":     "x128a3",
+    "atxmega128a3u":    "x128a3u",
+    "atxmega128a4":     "x128a4",
+    "atxmega128a4u":    "x128a4u",
+    "atxmega128b1":     "x128b1",
+    "atxmega128b3":     "x128b3",
+    "atxmega128c3":     "x128c3",
+    "atxmega128d3":     "x128d3",
+    "atxmega128d4":     "x128d4",
+    "atxmega16a4":      "x16a4",
+    "atxmega16a4u":     "x16a4u",
+    "atxmega16c4":      "x16c4",
+    "atxmega16d4":      "x16d4",
+    "atxmega16e5":      "x16e5",
+    "atxmega192a1":     "x192a1",
+    "atxmega192a3":     "x192a3",
+    "atxmega192a3u":    "x192a3u",
+    "atxmega192c3":     "x192c3",
+    "atxmega192d3":     "x192d3",
+    "atxmega256a1":     "x256a1",
+    "atxmega256a3":     "x256a3",
+    "atxmega256a3b":    "x256a3b",
+    "atxmega256a3bu":   "x256a3bu",
+    "atxmega256a3u":    "x256a3u",
+    "atxmega256c3":     "x256c3",
+    "atxmega256d3":     "x256d3",
+    "atxmega32a4":      "x32a4",
+    "atxmega32a4u":     "x32a4u",
+    "atxmega32c4":      "x32c4",
+    "atxmega32d4":      "x32d4",
+    "atxmega32e5":      "x32e5",
+    "atxmega384c3":     "x384c3",
+    "atxmega384d3":     "x384d3",
+    "atxmega64a1":      "x64a1",
+    "atxmega64a1u":     "x64a1u",
+    "atxmega64a3":      "x64a3",
+    "atxmega64a3u":     "x64a3u",
+    "atxmega64a4":      "x64a4",
+    "atxmega64a4u":     "x64a4u",
+    "atxmega64b1":      "x64b1",
+    "atxmega64b3":      "x64b3",
+    "atxmega64c3":      "x64c3",
+    "atxmega64d3":      "x64d3",
+    "atxmega64d4":      "x64d4",
+    "atxmega8e5":       "x8e5"
+}
+
+def getMcuForDevice(device_id):
+    d = device_id.copy()
+    # Search for key by truncating type
+    while(True):
+        LOGGER.debug("Searching for '%s'", d.string)
+        if d.string in avrdude_mcu_map:
+            return avrdude_mcu_map[d.string]
+        if len(d.type) == 0: break;
+        d.set("type", d.type[:-1])
+    LOGGER.warning("Avrdude mcu not found for '%s'!", device_id.string)
+    # otherwise return nothing
+    return None
```

Comment:
This file is not needed for Cortex-M based devices, this is only to map the device to the Avrdude name.